crates/oxc_minifier/src/ast_passes/peephole_remove_dead_code.rs

```diff
@@ -290,4 +309,35 @@ mod test {
         fold("for(;undefined;) foo()", "");
         fold("for(;'';) foo()", "");
     }
+
+    #[test]
+    fn test_object_literal() {
+        fold("({})", "");
+        fold("({a:1})", "");
```

Comment:
i'm looking to learn here 🙂 
```
({ foo: bar.baz })
```
shouldn't be folded is that correct (`bar.baz` could be a getter) ?

Reply:
> `a[b]` could trigger a getter or setter, and have side effects. However, we always assume it does not (even though it's unsound) because the code size cost of assuming all GETELEM nodes have side effects is unacceptable.

This is introduced from Google Closure Compiler's comments.

> shouldn't be folded is that correct (bar.baz could be a getter) ?

It makes sense, but this PR doesn't affect your given code.

Reply:
We need to compress it to `baz.baz`, but I haven’t implemented these non-literal scenarios yet.

Reply:
cool, thanks!